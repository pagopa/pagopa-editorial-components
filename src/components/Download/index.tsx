import { type CommonProps } from 'types/components';
import EContainer from '../EContainer';
import { Button, Typography, Divider, Stack } from '@mui/material';
import DownloadIcon from '@mui/icons-material/Download';
import { sendTheme } from '../../types/SendTheme';

interface IDownloadItem {
  label: string;
  fileName?: string;
  href: string;
}

export interface IDownload extends CommonProps {
  title: string;
  subtitle?: string;
  type?: 'button' | 'link';
  items: IDownloadItem[];
  itemsAlignment?: 'center' | 'left' | 'right';
}

const alignMap = new Map();
alignMap.set('left', 'flex-start');
alignMap.set('center', 'center');
alignMap.set('right', 'flex-end');

const Download = (props: IDownload) => {
  const {
    title,
    subtitle,
    items,
    theme,
    type,
    itemsAlignment = 'left',
  } = props;
  const backgroundColor =
    theme === 'dark' ? sendTheme.dark.backgroundColor : 'background.paper';
  const textColor = theme === 'dark' ? 'primary.contrastText' : 'text.primary';
  const isTypeButton = type === 'button';
  return (
    <EContainer direction="column" background={backgroundColor} py={10}>
      <Typography variant="h2" mb={1} color={textColor}>
        {title}
      </Typography>
      {subtitle && (
        <Typography variant="body1" mb={1} color={textColor}>
          {subtitle}
        </Typography>
      )}
      <Divider />
      <Stack alignItems={alignMap.get(itemsAlignment)}>
        {items.map((item, index) => (
          <Button
            key={`${item.label}-${index}`}
            href={item.href}
            variant={isTypeButton ? 'outlined' : 'text'}
            startIcon={<DownloadIcon />}
            color={theme === 'dark' ? 'negative' : 'primary'}
            download={item.fileName ?? ''}
            sx={{
              marginTop: isTypeButton ? '32px' : '8px',
              paddingLeft: isTypeButton ? 'inherith' : '6px',
            }}
          >
            {item.label}
          </Button>
        ))}
      </Stack>
    </EContainer>
  );
};

export default Download;
