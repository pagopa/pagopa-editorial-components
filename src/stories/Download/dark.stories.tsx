import { type ComponentMeta, type ComponentStory } from '@storybook/react';
import Download from '../../components/Download';
import pdf from '../../stories/assets/sample.pdf';

export default {
  component: Download,
  title: 'Download/dark',
  argTypes: {},
} as ComponentMeta<typeof Download>;

const Template: ComponentStory<typeof Download> = (args) => (
  <Download {...args} />
);

export const DarkButtons = Template.bind({});

DarkButtons.args = {
  theme: 'dark',
  type: 'button',
  title: 'Download Area Lorem Ipsum',
  subtitle:
    'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.',
  items: [
    {
      label: 'Download 1 Lorem Ipsum',
      fileName: 'sample_1.pdf',
      href: pdf,
    },
    {
      label: 'Download 2 Lorem Ipsum',
      fileName: 'sample_2.pdf',
      href: pdf,
    },
    {
      label: 'Download 3 Lorem Ipsum',
      fileName: 'sample_3.pdf',
      href: pdf,
    },
  ],
};

export const DarkLinks = Template.bind({});

DarkLinks.args = {
  theme: 'dark',
  type: 'link',
  title: 'Download Area Lorem Ipsum',
  subtitle:
    'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.',
  items: [
    {
      label: 'Download 1 Lorem Ipsum',
      fileName: 'sample_1.pdf',
      href: pdf,
    },
    {
      label: 'Download 2 Lorem Ipsum',
      fileName: 'sample_2.pdf',
      href: pdf,
    },
    {
      label: 'Download 3 Lorem Ipsum',
      href: pdf,
    },
  ],
};
