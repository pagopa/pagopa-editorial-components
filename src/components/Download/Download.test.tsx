import { ThemeProvider, createTheme } from '@mui/material';
import Download from './index';
import '@testing-library/jest-dom';
import { theme } from '../../../.storybook/theme';

/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';

const mainTheme = createTheme({
  ...theme,
});

it('renders correctly with light buttons', () => {
  const tree = renderer
    .create(
      <Download
        items={[
          {
            fileName: 'sample_1.pdf',
            href: 'static/media/sample.ac0398d4.pdf',
            label: 'Download 1 Lorem Ipsum',
          },
          {
            fileName: 'sample_2.pdf',
            href: 'static/media/sample.ac0398d4.pdf',
            label: 'Download 2 Lorem Ipsum',
          },
          {
            fileName: 'sample_3.pdf',
            href: 'static/media/sample.ac0398d4.pdf',
            label: 'Download 3 Lorem Ipsum',
          },
        ]}
        subtitle="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa."
        theme="light"
        title="Download Area Lorem Ipsum"
        type="button"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with light links', () => {
  const tree = renderer
    .create(
      <Download
        items={[
          {
            fileName: 'sample_1.pdf',
            href: 'static/media/sample.ac0398d4.pdf',
            label: 'Download 1 Lorem Ipsum',
          },
          {
            fileName: 'sample_2.pdf',
            href: 'static/media/sample.ac0398d4.pdf',
            label: 'Download 2 Lorem Ipsum',
          },
          {
            href: 'static/media/sample.ac0398d4.pdf',
            label: 'Download 3 Lorem Ipsum',
          },
        ]}
        subtitle="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa."
        theme="light"
        title="Download Area Lorem Ipsum"
        type="link"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with dark buttons', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={mainTheme}>
        <Download
          items={[
            {
              fileName: 'sample_1.pdf',
              href: 'static/media/sample.ac0398d4.pdf',
              label: 'Download 1 Lorem Ipsum',
            },
            {
              fileName: 'sample_2.pdf',
              href: 'static/media/sample.ac0398d4.pdf',
              label: 'Download 2 Lorem Ipsum',
            },
            {
              fileName: 'sample_3.pdf',
              href: 'static/media/sample.ac0398d4.pdf',
              label: 'Download 3 Lorem Ipsum',
            },
          ]}
          subtitle="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa."
          theme="dark"
          title="Download Area Lorem Ipsum"
          type="button"
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with dark links', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={mainTheme}>
        <Download
          items={[
            {
              fileName: 'sample_1.pdf',
              href: 'static/media/sample.ac0398d4.pdf',
              label: 'Download 1 Lorem Ipsum',
            },
            {
              fileName: 'sample_2.pdf',
              href: 'static/media/sample.ac0398d4.pdf',
              label: 'Download 2 Lorem Ipsum',
            },
            {
              href: 'static/media/sample.ac0398d4.pdf',
              label: 'Download 3 Lorem Ipsum',
            },
          ]}
          subtitle="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa."
          theme="dark"
          title="Download Area Lorem Ipsum"
          type="link"
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
