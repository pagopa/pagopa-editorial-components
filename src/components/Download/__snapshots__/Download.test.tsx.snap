// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with light buttons 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-a3n8s1{padding-top:80px;padding-bottom:80px;background-color:#fff;position:relative;overflow:hidden;}@media (min-width:0px){.css-a3n8s1{padding-left:32px;padding-right:32px;}}",
      }
    }
    data-emotion="css a3n8s1"
  />,
  <section
    className="MuiBox-root css-a3n8s1"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1p4tas-MuiContainer-root{width:100%;margin-left:auto;box-sizing:border-box;margin-right:auto;display:block;}@media (min-width:1200px){.css-1p4tas-MuiContainer-root{max-width:1200px;}}",
        }
      }
      data-emotion="css 1p4tas-MuiContainer-root"
    />
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1vpwcmr-MuiGrid-root{box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;width:100%;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.css-1vpwcmr-MuiGrid-root>.MuiGrid-item{max-width:none;}",
          }
        }
        data-emotion="css 1vpwcmr-MuiGrid-root"
      />
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1hdauvf-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:300;font-size:3.75rem;line-height:1.2;letter-spacing:-0.00833em;margin-bottom:8px;color:rgba(0, 0, 0, 0.87);}",
            }
          }
          data-emotion="css 1hdauvf-MuiTypography-root"
        />
        <h2
          className="MuiTypography-root MuiTypography-h2 css-1hdauvf-MuiTypography-root"
        >
          Download Area Lorem Ipsum
        </h2>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1ow4zxr-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;margin-bottom:8px;color:rgba(0, 0, 0, 0.87);}",
            }
          }
          data-emotion="css 1ow4zxr-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ow4zxr-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.
        </p>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-9mgopn-MuiDivider-root{margin:0;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;border-width:0;border-style:solid;border-color:rgba(0, 0, 0, 0.12);border-bottom-width:thin;}",
            }
          }
          data-emotion="css 9mgopn-MuiDivider-root"
        />
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1a69w1n-MuiStack-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;}",
            }
          }
          data-emotion="css 1a69w1n-MuiStack-root"
        />
        <div
          className="MuiStack-root css-1a69w1n-MuiStack-root"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1y73n7a-MuiButton-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(25, 118, 210, 0.5);color:#1976d2;margin-top:32px;padding-left:inherith;}.css-1y73n7a-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);border:1px solid #1976d2;}@media (hover: none){.css-1y73n7a-MuiButton-root:hover{background-color:transparent;}}.css-1y73n7a-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
              }
            }
            data-emotion="css 1y73n7a-MuiButton-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-gbk5jz-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(25, 118, 210, 0.5);color:#1976d2;margin-top:32px;padding-left:inherith;}.css-gbk5jz-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-gbk5jz-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-gbk5jz-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-gbk5jz-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);border:1px solid #1976d2;}@media (hover: none){.css-gbk5jz-MuiButtonBase-root-MuiButton-root:hover{background-color:transparent;}}.css-gbk5jz-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
              }
            }
            data-emotion="css gbk5jz-MuiButtonBase-root-MuiButton-root"
          />
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-gbk5jz-MuiButtonBase-root-MuiButton-root"
            download="sample_1.pdf"
            href="static/media/sample.ac0398d4.pdf"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1d6wzja-MuiButton-startIcon{display:inherit;margin-right:8px;margin-left:-4px;}.css-1d6wzja-MuiButton-startIcon>*:nth-of-type(1){font-size:20px;}",
                }
              }
              data-emotion="css 1d6wzja-MuiButton-startIcon"
            />
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
                  }
                }
                data-emotion="css i4bv87-MuiSvgIcon-root"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
            </span>
            Download 1 Lorem Ipsum
          </a>
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-gbk5jz-MuiButtonBase-root-MuiButton-root"
            download="sample_2.pdf"
            href="static/media/sample.ac0398d4.pdf"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
            </span>
            Download 2 Lorem Ipsum
          </a>
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-gbk5jz-MuiButtonBase-root-MuiButton-root"
            download="sample_3.pdf"
            href="static/media/sample.ac0398d4.pdf"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
            </span>
            Download 3 Lorem Ipsum
          </a>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`renders correctly with light links 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-a3n8s1{padding-top:80px;padding-bottom:80px;background-color:#fff;position:relative;overflow:hidden;}@media (min-width:0px){.css-a3n8s1{padding-left:32px;padding-right:32px;}}",
      }
    }
    data-emotion="css a3n8s1"
  />,
  <section
    className="MuiBox-root css-a3n8s1"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1p4tas-MuiContainer-root{width:100%;margin-left:auto;box-sizing:border-box;margin-right:auto;display:block;}@media (min-width:1200px){.css-1p4tas-MuiContainer-root{max-width:1200px;}}",
        }
      }
      data-emotion="css 1p4tas-MuiContainer-root"
    />
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1vpwcmr-MuiGrid-root{box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;width:100%;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.css-1vpwcmr-MuiGrid-root>.MuiGrid-item{max-width:none;}",
          }
        }
        data-emotion="css 1vpwcmr-MuiGrid-root"
      />
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1hdauvf-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:300;font-size:3.75rem;line-height:1.2;letter-spacing:-0.00833em;margin-bottom:8px;color:rgba(0, 0, 0, 0.87);}",
            }
          }
          data-emotion="css 1hdauvf-MuiTypography-root"
        />
        <h2
          className="MuiTypography-root MuiTypography-h2 css-1hdauvf-MuiTypography-root"
        >
          Download Area Lorem Ipsum
        </h2>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1ow4zxr-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;margin-bottom:8px;color:rgba(0, 0, 0, 0.87);}",
            }
          }
          data-emotion="css 1ow4zxr-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ow4zxr-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.
        </p>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-9mgopn-MuiDivider-root{margin:0;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;border-width:0;border-style:solid;border-color:rgba(0, 0, 0, 0.12);border-bottom-width:thin;}",
            }
          }
          data-emotion="css 9mgopn-MuiDivider-root"
        />
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1a69w1n-MuiStack-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;}",
            }
          }
          data-emotion="css 1a69w1n-MuiStack-root"
        />
        <div
          className="MuiStack-root css-1a69w1n-MuiStack-root"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-ngsttt-MuiButton-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 8px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;margin-top:8px;padding-left:6px;}.css-ngsttt-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);}@media (hover: none){.css-ngsttt-MuiButton-root:hover{background-color:transparent;}}.css-ngsttt-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);}",
              }
            }
            data-emotion="css ngsttt-MuiButton-root"
          />
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zj1dvw-MuiButtonBase-root-MuiButton-root"
            download="sample_1.pdf"
            href="static/media/sample.ac0398d4.pdf"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1d6wzja-MuiButton-startIcon{display:inherit;margin-right:8px;margin-left:-4px;}.css-1d6wzja-MuiButton-startIcon>*:nth-of-type(1){font-size:20px;}",
                }
              }
              data-emotion="css 1d6wzja-MuiButton-startIcon"
            />
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
                  }
                }
                data-emotion="css i4bv87-MuiSvgIcon-root"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
            </span>
            Download 1 Lorem Ipsum
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-8je8zh-MuiTouchRipple-root{overflow:hidden;pointer-events:none;position:absolute;z-index:0;top:0;right:0;bottom:0;left:0;border-radius:inherit;}",
                }
              }
              data-emotion="css 8je8zh-MuiTouchRipple-root"
            />
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zj1dvw-MuiButtonBase-root-MuiButton-root"
            download="sample_2.pdf"
            href="static/media/sample.ac0398d4.pdf"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
            </span>
            Download 2 Lorem Ipsum
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zj1dvw-MuiButtonBase-root-MuiButton-root"
            download=""
            href="static/media/sample.ac0398d4.pdf"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
            </span>
            Download 3 Lorem Ipsum
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </section>,
]
`;
